lib/core/src/Cardano/Wallet.hs

```diff
@@ -439,6 +451,16 @@ newWalletLayer tracer block0 feePolicy db nw tl = do
     _listWallets =
         fmap (\(PrimaryKey wid) -> wid) <$> DB.listWallets db
 
+    _listTransactions
+        :: WalletId
+        -> ExceptT ErrNoSuchWallet IO [(Tx t, TxMeta)]
+    _listTransactions wid = liftIO $
+        L.sortOn slotIdDescending
+            . map snd
+            . Map.toList <$> DB.readTxHistory db (PrimaryKey wid)
```

Comment:
I think sorting is very much a job for the DBLayer.

Can we
1. Add a comment on `DBLayer.readTxHistory` saying that the Map short be sorted descending by slot.
2. Adjust the DBLayer backends accordingly.

Reply:
@KtorZ (asking here instead) you said

> Either we gotta change the type (which I think isn't really a good idea), or do the sorting in-memory in the handler itself.

Could you elaborate on why you didn't think it was a good idea?

I'm still familiarising myself with the Sqlite schema (+ sqlite). Could we not either:

1. Just make `txHistoryFromEntity` return a sorted `[(W.Tx t, W.TxMeta)]` given a sorted `[W.TxMeta]` instead of returning a `Map`? It seems like we currently only use `readTxHistory` in `WalletLayer._listAddresses`.

2. Use only one select (from three tables at once `tx_meta`, `tx_in` and `tx_out`)… somehow? (not sure if this is possible)

Reply:
> why you didn't think it was a good idea?

Mainly because of the rippling effect and the amount of changes it involves in the code-base. If we've to go through this, then, we'll have too :)